### PR TITLE
resolves #32: Allow the status to be exported as json, so that it can…

### LINKED ISSUE #32: Allow the status to be exported as json
We would like to use the RDE env in a CI / CD environment, where we need to know if the environment is available before installing a next package. 

The aem:rde:status is a textual description, it is non-parseable for an automated system. 

Feature request:
- Add a json output for the status command

### PR DESCRIPTION
**Pull request by royteeuwen**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -38,7 +38,7 @@ The plugin needs to be configured to point to an existing RDE environment as fol
 1. Run `aio login`
 2. Run `aio aem:rde` for general help.
 3. Run `aio aem:rde:install --help ` for help about a specific command.
-4. Run `aio aem:rde:status` to see if the configured environment can be accessed.
+4. Run `aio aem:rde:status --help` to see if the configured environment can be accessed.
```
Comment (jsedding) [inline README.md]: That change doesn't look right?

File: src/commands/aem/rde/status.js [hunk 2]
```diff
@@ -57,12 +67,53 @@ class StatusCommand extends BaseCommand {
...
+        loadAllArtifacts(cloudSdkAPI)
+      );
+
+      const grouped = groupArtifacts(status.items);
+
+      const result = {
+        programId: this._programId,
+        environmentId: this._environmentId,
+        status: status.status,
+      }
```
Comment (jsedding) [inline src/commands/aem/rde/status.js:84]: This block should go into the `else` block.

File: src/commands/aem/rde/status.js [hunk 3]
```diff
@@ -57,12 +67,53 @@ class StatusCommand extends BaseCommand {
...
+
+      if (status.error) {
+        result.statusText = status.BaseCommand;
+      } else {
+        result.author = {
+          'osgi-bundle': grouped.author['osgi-bundle'],
+          'osgi-config': grouped.publish['osgi-config']
```
Comment (jsedding) [inline src/commands/aem/rde/status.js]: I would give the fields in the new JSON plural names. E.g. `osgi-bundles` or `osgiBundles` etc.

**Review by jsedding — CHANGES_REQUESTED** [review]
Looks good in general. Could you please look at my comments? Also, it would be awesome if  you could add a simple test case!

Ideally, we would also re-use more code between `printAsText` and `printAsJson`. The main difference seems to be the "spinner" and the output being written progressively in the case of `printAsText`. Maybe these differences could be factored out via a callback. WDYT?

File: src/commands/aem/rde/status.js (hunk 2 above)
Comment (royteeuwen) [reply src/commands/aem/rde/status.js:84]: This way you will also get the program id and environment id when there is an error, could be useful to see if you misconfigured something there

File: README.md (hunk 1 above)
Comment (royteeuwen) [reply README.md]: Indeed, removed

**Comment by royteeuwen** [comment]
> Ideally, we would also re-use more code between `printAsText` and `printAsJson`. The main difference seems to be the "spinner" and the output being written progressively in the case of `printAsText`. Maybe these differences could be factored out via a callback. WDYT?

I tried to do that, but seeing as the flow is completely different, it's pretty hard. The printing already starts from the beginning for the printAsText while the printAsJson has no output until the end

**Comment by royteeuwen** [comment]
> Looks good in general. Could you please look at my comments? Also, it would be awesome if you could add a simple test case!

done

**Comment by jsedding** [comment]
Thanky for the contribution @royteeuwen!

**Pull request merged**